### PR TITLE
README: add section explaining motivation and scope

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,19 @@
 
 **Babelfish is a self-hosted server for source code parsing.** Babelfish can parse any file, in any language, extract an ASTs (Abstract Syntax Tree) from it, and convert it to a [**UAST**](./uast/specification.md) (Universal Abstract Syntax Tree).
 
+## Motivation and Scope
+
+Babelfish was born as a solution for massive code analysis. The goal: analyzing
+all source code from every repository, for every revision.
+
+Current **scope is parsing of single files in any programming language**
```
Comment (abeaumont) [inline README.md]: Maybe better `Current scope is **parsing single files in any programming language**`?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,19 @@
...
+that produces a universal abstract syntax tree. This helps us to run it at scale.
+
+This scope might expand in the future to full project analysis, where source code
+can be analyzed with its full context, an not just per-file. However, this would
```
Comment (abeaumont) [inline README.md]: I find this last sentence a bit overcomplicated. Maybe something along _Once Babelfish with its current scope is mature enough, we may consider starting such effort_.

File: README.md (hunk 1 above)
Comment (smola) [inline README.md]: :+1:

File: README.md (hunk 2 above)
Comment (smola) [inline README.md]: :+1:

File: README.md [hunk 3]
```diff
@@ -4,6 +4,18 @@
 
 **Babelfish is a self-hosted server for source code parsing.** Babelfish can parse any file, in any language, extract an ASTs (Abstract Syntax Tree) from it, and convert it to a [**UAST**](./uast/specification.md) (Universal Abstract Syntax Tree).
 
+## Motivation and Scope
+
+Babelfish was born as a solution for massive code analysis. The goal: analyzing
+all source code from every repository, for every revision.
```
Comment (alcortesm) [inline README.md]: add some scope:

```
all source code from every repository in the world, for every revision ever.
```

File: README.md [hunk 4]
```diff
@@ -4,6 +4,18 @@
...
+
+Current **scope is parsing single files in any programming language**
+that produces a universal abstract syntax tree. This helps us to run it at scale.
```
Comment (alcortesm) [inline README.md]: s/that produces/and produce/

File: README.md [hunk 5]
```diff
@@ -4,6 +4,18 @@
...
+
+This scope might expand in the future to full project analysis, where source code
+can be analyzed with its full context, an not just per-file. Once Babelfish with
+its current scope is mature enough, we may consider starting such effort.
```
Comment (alcortesm) [inline README.md]: Suggested rewording:

```
we will consider embarking on such an endeavour once we fulfil our current scope.
```
We have to find a better word for `scope`.

Comment (smola) [inline README.md]: `embarking on such an endeavour` sounds a bit pedantic to me...

File: README.md (hunk 4 above)
Comment (smola) [inline README.md]: :+1:

File: README.md (hunk 3 above)
Comment (smola) [inline README.md]: :+1:

File: README.md (hunk 5 above)
Comment (alcortesm) [inline README.md]: That's fine, I suggest this other version then:
```
Once we get closer to our current scope, we may consider starting such effort.
```

Comment (alcortesm) [inline README.md]: It is the `with` word the one bugging me.

Comment (smola) [inline README.md]: :+1:

**Pull request merged**